pkgs/applications/emulators/wine/base.nix

```diff
@@ -201,16 +200,15 @@ stdenv.mkDerivation ((lib.optionalAttrs (buildScript != null) {
     tests = { inherit (nixosTests) wine; };
   };
   meta = {
-    inherit version;
+    inherit version platforms;
     homepage = "https://www.winehq.org/";
     license = with lib.licenses; [ lgpl21Plus ];
     sourceProvenance = with lib.sourceTypes; [
       fromSource
       binaryNativeCode  # mono, gecko
     ];
     broken = stdenv.isDarwin && !supportFlags.mingwSupport;
-    description = if supportFlags.waylandSupport then "An Open Source implementation of the Windows API on top of OpenGL and Unix (with experimental Wayland support)" else "An Open Source implementation of the Windows API on top of X, OpenGL, and Unix";
-    platforms = if supportFlags.waylandSupport then (lib.remove "x86_64-darwin" prevPlatforms) else prevPlatforms;
```

Comment:
Did you mean to remove the `platforms` definition? Wayland isn’t supported on Darwin.

Reply:
It is possible to build `wine-wayland` without Wayland. (This line was originally checking for the "wayland" release)

If you meant "the evaluation will fail if `waylandSupport` is true because the dependencies don't exist on Darwin"... you are right, but that would also apply to other dependencies too.

Reply:
So, options in my mind:
- In `base.nix`, don't just check for `dependencySupport`, but check for `dependencySupport && stdenv.isRightOS`, so that the flag is ignored on the wrong operating system
  - This however causes the problem of having stuff like `wine-wayland` without Wayland on Darwin.
- In `base.nix`, add a list of `supportFlags` that aren't on Darwin: if any of these are true, disable Darwin support (which is the approach of before, but not just for Wayland)
  - This may make a Wine build accidentally incompatible with Darwin, and it might be hard to understand why. In `wine-packages.nix` you have to be careful on what is `= stdenv.isLinux`.

Reply:
The wine-wayland packages fail to build on Darwin. I’d like to avoid introducing unnecessary Hydra failures.

> So, options in my mind:
> 
> * In `base.nix`, don't just check for `dependencySupport`, but check for `dependencySupport && stdenv.isRightOS`, so that the flag is ignored on the wrong operating system
>   
>   * This however causes the problem of having stuff like `wine-wayland` without Wayland on Darwin.
> * In `base.nix`, add a list of `supportFlags` that aren't on Darwin: if any of these are true, disable Darwin support (which is the approach of before, but not just for Wayland)

Silently disabling requested support for a feature would be a bad experience for users. It would be better add Darwin to `badPlatforms` for flags known to fail to build on that platform.

>   * This may make a Wine build accidentally incompatible with Darwin, and it might be hard to understand why. In `wine-packages.nix` you have to be careful on what is `= stdenv.isLinux`.

Some of the flags set in `wine-packages.nix` may result in working builds on Darwin but don’t make sense being enabled by default (such as `gtkSupport` and `x11Support`). If a user really wants to enable one of them anyway, they can override one of the Wine packages to enable it.